contracts/misc/AirdropFlashLoanReceiverV3.sol

```diff
@@ -0,0 +1,220 @@
+
+    require(vars.airdropContract != address(0), "invalid airdrop contract address");
+    require(vars.airdropParams.length >= 4, "invalid airdrop parameters");
+
+    // allow operator transfer borrowed nfts back to bnft
+    IERC721Upgradeable(nftAsset).setApprovalForAll(operator, true);
+
+    // call project aidrop contract
+    AddressUpgradeable.functionCall(vars.airdropContract, vars.airdropParams, "call airdrop method failed");
+
+    vars.airdropKeyHash = getClaimKeyHash(initiator, nftAsset, nftTokenIds, params);
+    airdropClaimRecords[vars.airdropKeyHash] = true;
+
+    // transfer airdrop tokens to borrower
+    for (uint256 typeIndex = 0; typeIndex < vars.airdropTokenTypes.length; typeIndex++) {
+      require(vars.airdropTokenAddresses[typeIndex] != address(0), "invalid airdrop token address");
+
+      if (vars.airdropTokenTypes[typeIndex] == 1) {
+        // ERC20
+        vars.airdropBalance = IERC20Upgradeable(vars.airdropTokenAddresses[typeIndex]).balanceOf(address(this));
+        if (vars.airdropBalance > 0) {
+          IERC20Upgradeable(vars.airdropTokenAddresses[typeIndex]).transfer(initiator, vars.airdropBalance);
+        }
+      } else if (vars.airdropTokenTypes[typeIndex] == 2) {
+        // ERC721 with Enumerate
+        vars.airdropBalance = IERC721Upgradeable(vars.airdropTokenAddresses[typeIndex]).balanceOf(address(this));
+        for (uint256 i = 0; i < vars.airdropBalance; i++) {
+          vars.airdropTokenId = IERC721EnumerableUpgradeable(vars.airdropTokenAddresses[typeIndex]).tokenOfOwnerByIndex(
+            address(this),
+            0
+          );
+          IERC721EnumerableUpgradeable(vars.airdropTokenAddresses[typeIndex]).safeTransferFrom(
+            address(this),
+            initiator,
+            vars.airdropTokenId
+          );
+        }
+      } else if (vars.airdropTokenTypes[typeIndex] == 3) {
+        // ERC1155
+        vars.airdropBalance = IERC1155Upgradeable(vars.airdropTokenAddresses[typeIndex]).balanceOf(
+          address(this),
+          vars.airdropTokenIds[typeIndex]
+        );
+        IERC1155Upgradeable(vars.airdropTokenAddresses[typeIndex]).safeTransferFrom(
+          address(this),
+          initiator,
+          vars.airdropTokenIds[typeIndex],
+          vars.airdropBalance,
+          new bytes(0)
+        );
+      } else if (vars.airdropTokenTypes[typeIndex] == 4) {
+        // ERC721 without Enumerate
+        IERC721EnumerableUpgradeable(vars.airdropTokenAddresses[typeIndex]).safeTransferFrom(
+          address(this),
+          initiator,
+          vars.airdropTokenIds[typeIndex]
+        );
+      }
+    }
+
+    return true;
+  }
+
+  function executeClaim(address claimContract, bytes calldata claimParams) external nonReentrant onlyOwner {
+    require(claimContract != address(0), "invalid contract address");
+    require(claimParams.length >= 4, "invalid parameters");
```

Comment:
Why is the claimParams length greater than 4 required here?

Reply:
bcs mehod id is 4 bytes, so param >= 4